src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5581,18 +5589,14 @@ public boolean isValid(int timeout) throws SQLException {
             // If it fails (throws an exception), then the connection is not valid.
             // If a timeout was provided, execution throws an "query timed out" exception
             // if the query fails to execute in that time.
-            stmt.executeQueryInternal("SELECT 1");
-            stmt.close();
+            stmt.execute("SELECT 1");
             isValid = true;
         } catch (SQLException e) {
             // Do not propagate SQLExceptions from query execution or statement closure.
             // The connection is considered to be invalid if the statement fails to close,
             // even though query execution succeeded.
             connectionlogger.fine(toString() + " Exception checking connection validity: " + e.getMessage());
-        } finally {
-            if (null != stmt && !stmt.isClosed()) {
-                stmt.close();
-            }
+            isValid = false;
```

Comment:
Nit-picky, but unnecessary line? It's by default false.

Reply:
Here its needed, coz if `stmt.close()` fails.. we still want to invalidate.. or I can reverse the logic as in your PR..